code/espurna/relay.ino

```diff
@@ -894,60 +903,76 @@ String _relayFriendlyName(unsigned char i) {
     return res;
 }
 
+
 void _relayWebSocketSendRelays(JsonObject& root) {
-    JsonObject& relays = root.createNestedObject("relayConfig");
+    JsonObject& module = root.createNestedObject("relay");
+    JsonObject& config = module.createNestedObject("config");
 
-    relays["size"] = relayCount();
-    relays["start"] = 0;
+    JsonArray& schema = config.createNestedArray("_schema");
+    schema.add("GPIO");
+    schema.add("type");
+    //schema.add("resetGPIO"); //This is not needed
+    schema.add("name");
+    schema.add("boot");
+    schema.add("pulse");
+    schema.add("time");
+
+    schema.add("dblDl");
+    schema.add("lngDl");
+    schema.add("lngLngDl");
```

Comment:
re my previous comment, clicks & click delays are button settings. I'll add the specific part to the button code then

Reply:
Hmm didn't think about that since the relationship between buttons and relays is almost always 1/1 but there might be cases where buttons don't have a specific relay or vice versa..
In which case it might be a good idea to expose the settings on the button module and to do the relation later on

In the future, that could open the door for a device module where the buttons/leds/relays GPIO, manufacturer and device could be configured. That would allow for non hardcoded generic builds in pio where you could configure your buttons/relays/led on different GPIOs, edit the manufacturer and device freely where one build fits a lot of buttons/relays/leds only devices and you can just have a list/database of configurations online depending on the device (Eg: sonoff 1ch, 2ch, 3ch, touch, sv etc) to just import to the device

Reply:
I think I wanted to do buttons in dev branch first :>
Note on the https://github.com/xoseperez/espurna/pull/1933/commits/b352284ea9f3c676b5a35805d8d8716c1583c93c - .ino should preferably be using constexpr funcs that return preprocessor vars, that way you already know the type.

But I agree on board ID thing, I don't think I want that after all. Same with migrate, only keeping that for now to reference keyvalues that were used before

Reply:
Yep I changed all that already :p
I'll be organizing some more stuff (a few settings to rename) and then start testing on a device to finish up the vue app,

For the board id it's not used anywhere and we already have the MANUFACTURER and DEVICE info, for the migration we can keep it (though it may take some build space)
I'll need to add a new config version and migrate some settings anyway so I'll use it

I'll add the buttons settings under the leds tabs and rename it Leds/Buttons
Avoid that clustered mess that is relay

Reply:
I tried some board stuff in the #1680, but was still different from the old v2 approach. Main idea was to support device 'sets' that consisted of key-value pairs and swap config based on that build-time ID. migrate.ino was also safe-guarded against overwriting, like it does right now. 
Headers are finally fixed, so sourcing config/all.h with any external app is much more easier than it was in that PR.

My only concern is to not get too broad with modifications (while staying on the topic of Vue UI, see the title :). But, if that helps with an overall design of the UI, sure; better integrate the stuff now when it is less pain to change.

And as I was saying - I may break add the same thing to dev in the meantime which will probably involve manual merging of stuff with configuration. Mostly how things are chained together and supporting indexed stuff at build time.

Reply:
Hmm for the migrate, that mean it's not implemented yet since from what I see none of the settings set there are actually used yet..
(Eg relayGPIO, relayType, btnGPIO)

And it was actually an attempt to try what I suggested with the database of devices/settings except here it's hardcoded in the binary

Reply:
Nah, forget about migrate, we already have device settings in the hardware.h + key association is implemented in each module separately instead of having it all in one place. It was a prototype, but there is no actual point of having it anymore.
The general idea from v2 & pr is to keep key-value pairs for each device, based on current MANUFACTURER_DEVICE flag (either all at once, or limited to user-specified set).

As this PR is still wip there is no point of really restricting yourself, but just keep in mind that I'd prefer to merge settings and API changes beforehand. Let's just see what works